### ISSUE TITLE
Get inherited members of a class or parent class declaration

### ISSUE THREAD

**Issue by mnmtanish** [dsherret/ts-simple-ast#103]
A method to get all inherited members of a class or to get the parent class declaration can be quite useful for code generating tools.

**Comment by mnmtanish** [comment]
I'm going to implement this feature outside ts-simple-ast anyways but if it can be useful, I'll send a PR.

**Comment by dsherret** [comment]
Cool thanks! Probably getting the base class is sufficient.

I have thought about this before and should be easy to implement, but I was wondering how to handle mixins.

```typescript
export type Constructable = new (...args: any[]) => {};

export function Timestamped<BC extends Constructable>(Base: BC) {
    return class extends Base {
        timestamp = new Date();
    };
}

class Point {
    x: number;
    y: number;
    
    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }
}

// case 1
class SpecialPoint extends Timestamped(Point) {
    z: number;

    constructor(x: number, y: number, z: number) {
        super(x, y);
        this.z = z;
    }
}

// case 2
const OtherPointBase = Timestamped(Point);
class OtherPoint extends OtherPointBase {
}
```

If you implement something that just handles the common scenario that should be perfect though! I can work out handling mixins later.

**Comment by yang** [comment]
FWIW, TypeDoc's internals do provide types of inherited members. TypeDoc actually has quite a bit of useful introspection code, which may be very relevant to ts-simple-ast.

**Comment by dsherret** [comment]
I added two methods `.getBaseTypes()` and `.getBaseClass().

If you don't have to worry about mixins, then use `.getBaseClass()`... if you do, then use `.getBaseTypes()`. This will be available in v5.4 (releasing soon)
